Add Raspberry Pi Getting Started guide (#101)
Sensirion/embedded-sgp: docs/sgp40/raspisetup.rst

SGP40 on Raspberry Pi
=====================

This document explains how to set the SGP40 up to run with a Raspberry
Pi. This became very easy with the new SGP40 since it can run with a
logic level of 3.3v which matches the one the Raspberry Pi has.

Getting started
---------------

Connecting the Sensor
~~~~~~~~~~~~~~~~~~~~~

| Your Sensor should have four different connectors: VCC, GND, SDA, SCL.
Please refere to other documentation of the Sensor to find out which
connector is which on your specific modelle.
| Use the following pins to connect your SGP40:

+-----------+------------------+
| *SGP40*   | *Raspberry Pi*   |
+===========+==================+
| VCC       | Pin 1            |
+-----------+------------------+
| GND       | Pin 6            |
+-----------+------------------+
| SDA       | Pin 3            |
+-----------+------------------+
| SCL       | Pin 5            |
+-----------+------------------+

|RaspberryPi Pinout|

Raspberry Pi
~~~~~~~~~~~~

-  **Install the newest Raspberry Pi OS on to your Raspberry Pi. If you
   don't know how to do that, there is a guide on the Raspberry Pi

>    website**

Author: The bold is because the formatting automatically makes bullet points with sub points bold but those without them not. So to keep the formatting consistent I made them bold manually.

   (https://projects.raspberrypi.org/en/projects/raspberry-pi-setting-up).
-  Enable the I2C interface in the Raspberry Pi config.
    -  Open a terminal.
    -  Run ``sudo raspi-config``.
    -  Select "5 Interfacing Options Configure connections to peripherals".
    -  Select "P5 I2C Enable/Disable automatic loading of I2C kernel module".
    -  Select "Yes" when questioned "Would you like the ARM I2C interface to be enabled?".
    -  The Raspberry Pi should respond with "The ARM I2C interface is enabled". Confirm with "Ok"
-  Download driver.
    -  Go to the Sensirion SGP Driver Release page (https://github.com/Sensirion/embedded-sgp/releases) and download the ``sgp40-VERSION.zip`` file.
    -  Unzip the file into the directory where you want to install the driver (Run: ``unzip sgp40-VERSION.zip`` ).
-  Adapt the driver to work with Raspberry Pi.

>     -  We use the linux\_user\_space implementation of the hw\_i2c to run the driver on the Raspberry Pi. To use the in the project provided sample implementation navigate to the ``./hw_i2c/sample-implementations/linux_user_space`` direcory.

Reviewer: I'd use \`\` for all the path references.

    -  Copy the file ``sensirion_hw_i2c_implementation.c`` from there and replace the file named the same directly in ``./hw_i2c`` or just run this command in the root directory of your project:
   ``cp ./hw_i2c/sample-implementations/linux_user_space/sensirion_hw_i2c_implementation.c ./hw_i2c/``.
-  Compile the driver.
    -  Run ``make`` in the root directory of your project.

Output:

::

    rm -f sgp40_example_usage
    cc -Os -Wall -fstrict-aliasing -Wstrict-aliasing=1 -Wsign-conversion -fPIC -I. -I. -I. -I./hw_i2c -o sgp40_example_usage ./sensirion_arch_config.h ./sensirion_i2c.h ./sensirion_common.h ./sensirion_common.c ./sgp_git_version.h ./sgp_git_version.c ./sgp40.h ./sgp40.c ./hw_i2c/sensirion_hw_i2c_implementation.c ./sgp40_example_usage.c

-  **To test the setup run the example code which was just compiled with**
   ``./sgp40_example_usage`` **.**

Output:

::

    SGP40 driver version 6.0.0-32-gc792a99
    SGP sensor probing successful
    SerialID: 00:00:02:37:BF:3D
    sraw: 30403
    sraw: 30414
    sraw: 30407
    sraw: 30403
    sraw: 30390
    sraw: 30379
    sraw: 30370
    sraw: 30383
    ...

Troubleshooting
---------------

SGP sensor probing failed

> ~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer: Maybe state the exact error: "SGP sensor probing failed" if I remember correctly.

Author: You don't ;)

> -  Check if you connected the sensor correctly, so all cables are fully

Reviewer 2: ```suggestion
-  Ensure that you connected the sensor correctly: all cables are fully
```

   pluged in and connected to the correct header.

> -  Check if you have enabled I2C on the Raspberry Pi. For this redo the

Reviewer 2: ```suggestion
-  Ensure that I2C is enabled on the Raspberry Pi. For this redo the
```

   steps on "Enable the I2C interface in the Raspberry Pi config" in the
   guide above.

> 

Reviewer 2: another common problem is permissions, maybe something along the lines "Ensure that your user account has read and write access to the i2c device. If it works with user root ("sudo ./sgp40_example_usage"), it's typically due to wrong permission settings."

SGP40 on Raspberry Pi
=====================

This document explains how to set up the SGP40 to run with a Raspberry Pi.  
Since the SGP40 runs at 3.3V, which matches the Raspberry Pi's logic level,  

> no level shifting is required on the i2c bus.Your sensor has the four different connectors: VCC, GND, SDA, SCL.

Reviewer: ```suggestion
no level shifting is required on the i2c bus. Your sensor has the four
different connectors: VCC, GND, SDA, SCL.
```
